modules/abstract-utxo/src/recovery/RecoveryProvider.ts

```diff
@@ -15,13 +15,14 @@ export interface RecoveryAccountData {
 /**
  * Factory for AddressApi & UtxoApi
  */
-export interface RecoveryProvider {
-  getUnspentsForAddresses(addresses: string[]): Promise<Unspent[]>;
+export interface RecoveryProvider<TNumber extends number | bigint = number> {
+  getUnspentsForAddresses(addresses: string[]): Promise<Unspent<TNumber>[]>;
   getAddressInfo(address: string): Promise<AddressInfo>;
   getTransactionHex(txid: string): Promise<string>;
+  getTransactionIO(txid: string): Promise<TransactionIO>;
 }
 
-export function forCoin(coinName: string, apiToken?: string): RecoveryProvider {
+export function forCoin(coinName: string, apiToken?: string): RecoveryProvider<number> {
```

Comment:
Making this return a TNumber RecoveryProvider proved to be a much larger refactor as it required more changes within each blockapi implementation